app/controllers/oauth_controller.rb

```diff
@@ -35,20 +35,45 @@ def authorize
            status: :bad_request,
            locals: { error_class: error.class, message: error.message }
   end
-  # rubocop:enable Metrics/MethodLength, Metrics/AbcSize
 
   def token
     authorization_grant = AuthorizationGrant.find_by(id: params[:code])
     TokenRequestValidatorService.call!(
       authorization_grant:, code_verifier: params[:code_verifier], grant_type: params[:grant_type]
     )
+
+    client_id = authorization_grant.client_id
+    access_token_expiration = 5.minutes.from_now
+    access_token_jti, access_token = OAuthTokenEncoderService.call(
+      client_id:,
+      expiration: access_token_expiration,
+      optional_claims: { user_id: authorization_grant.user_id }
+    ).deconstruct
+
+    refresh_token_jti, refresh_token = OAuthTokenEncoderService.call(
+      client_id:,
+      expiration: 14.days.from_now
```

Comment:
Not a huge deal, but i would put this in a constant to make it a bit clearer. You could also move `access_token_expiration` to a constant if you want

Reply:
Great suggestion, I'll add it to the OAuth config file.